src/modm/architecture/detect.hpp

```diff
@@ -190,23 +186,20 @@
 
 #if defined __X86__ || defined __i386__ || defined i386 || defined _M_IX86 || defined __386__ || defined __x86_64__ || defined _M_X64
```

Comment:
```suggestion
#ifdef __X86__ || defined __i386__ || defined i386 || defined _M_IX86 || defined __386__ || defined __x86_64__ || defined _M_X64
```

Is there a reason why you only replaced `#if defined` in some places?

Reply:
`#ifdef` doesn't allow for additional logic chained behind it.